src/test/java/org/opensearch/flowframework/workflow/DeleteConnectorStepTests.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.opensearch.action.delete.DeleteResponse;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.index.Index;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.common.CommonValue;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.ml.common.transport.connector.MLCreateConnectorResponse;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.verify;
+
+public class DeleteConnectorStepTests extends OpenSearchTestCase {
+    private WorkflowData inputData;
+
+    @Mock
+    MachineLearningNodeClient machineLearningNodeClient;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        MockitoAnnotations.openMocks(this);
+
+        inputData = new WorkflowData(Map.of(CommonValue.CONNECTOR_ID, "test"), "test-id", "test-node-id");
+    }
+
+    public void testDeleteConnector() throws IOException, ExecutionException, InterruptedException {
+
+        String connectorId = randomAlphaOfLength(5);
+        DeleteConnectorStep deleteConnectorStep = new DeleteConnectorStep(machineLearningNodeClient);
+
+        @SuppressWarnings("unchecked")
+        ArgumentCaptor<ActionListener<DeleteResponse>> actionListenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+
+        doAnswer(invocation -> {
+            String connectorIdArg = invocation.getArgument(0);
+            ActionListener<DeleteResponse> actionListener = invocation.getArgument(1);
+            ShardId shardId = new ShardId(new Index("indexName", "uuid"), 1);
+            DeleteResponse output = new DeleteResponse(shardId, connectorIdArg, 1, 1, 1, true);
+            actionListener.onResponse(output);
+            return null;
+        }).when(machineLearningNodeClient).deleteConnector(any(String.class), actionListenerCaptor.capture());
```

Comment:
Curious, what are we using the `actionListenerCaptor` for here? I don't see any assertions for the values we get from this

Reply:
The direct answer to your question is, "I just copied and pasted from the Create Connector test".

Which suffers from the same unused captor.  I'll try to address both.

Reply:
Looks like earlier tests using the captor relied on it to invoke the `onResponse()`, but these tests are using a `doAnswer()` approach instead.  So I'll remove the capturing.

Reply:
Thanks Dan for the changes, looks great :)